### PR TITLE
Feature/ntrnl  132 improve state files and data structure

### PR DESCRIPTION
**Pull request by Mouhajer-CO**
### JIRA link

Resolves [NTRNL-132](https://technologyprogramme.atlassian.net/browse/NTRNL-132)

### Change description

Improve state files and data structure.
Update Team dashboard 
Update docs 

### Work checklist

- [x] Tests added where applicable

### CODE REVIEW

File: test/src/scripts/main.spec.ts [hunk 1]
```diff
@@ -5,15 +5,15 @@ import { describe, expect, test, jest, afterEach, beforeEach } from '@jest/globa
...
 
 import { MOCK_ORGANIZATION } from '../../mock/repos_info';
-import { REPOS_FILE_PATH, TEAMS_FILE_PATH } from '../../../src/config';
+import { REPOS_FILE_PATH } from '../../../src/config';
 
 const spyConsoleError = jest.spyOn(console, 'error');
 
-const mockGetGitOrgData = getGitOrgData as jest.Mock;
-const mockSetTeamsData = setTeamsData as jest.Mock;
+const mockGeOrgData = getOrgData as jest.Mock;
```
Comment (Harrisman05) [inline test/src/scripts/main.spec.ts:15]: `mockGeOrgData` missing letter t --> `mockGetOrgData`

File: test/src/scripts/main.spec.ts [hunk 2]
```diff
@@ -5,15 +5,15 @@ import { describe, expect, test, jest, afterEach, beforeEach } from '@jest/globa
...
-import { getGitOrgData, setTeamsData } from "../../../src/utils/index";
+import { getOrgData, getTeamsData } from "../../../src/utils/index";
 
 import { MOCK_ORGANIZATION } from '../../mock/repos_info';
-import { REPOS_FILE_PATH, TEAMS_FILE_PATH } from '../../../src/config';
+import { REPOS_FILE_PATH } from '../../../src/config';
 
 const spyConsoleError = jest.spyOn(console, 'error');
 
-const mockGetGitOrgData = getGitOrgData as jest.Mock;
```
Comment (Harrisman05) [inline test/src/scripts/main.spec.ts:15]: Should `setOrgData()` inside `main.ts` also be mocked and checked that it's called?

File: test/src/utils/fs.spec.ts [hunk 3]
```diff
@@ -106,23 +108,28 @@ describe("UTILS fs tests suites", () => {
     describe("cloneRepos(...)", () => {
```
Comment (Harrisman05) [inline test/src/utils/fs.spec.ts:108]: I would move this `"cloneRepos(...)` describe block to top of testing scope to match the structure of functions in `fs.ts`

Comment (Mouhajer-CO) [reply test/src/utils/fs.spec.ts:108]: Thanks for looking into It but it is not important the order of the nested `describe()` function

File: test/src/utils/index.spec.ts [hunk 4]
```diff
@@ -117,165 +125,200 @@ describe("UTILS Index tests suites", () => {
...
+    describe("getInfo(...)", () => {
 
         afterEach(() => {
             // Reset object data
-            ORG_DATA["members"] = []; ORG_DATA["repos"] = []; ORG_DATA["teams"] = [];
+            TMP_DATA["repos"]["list"] = [];
         });
 
-        test('should return empty list', async () => {
+        test('should TMP_DATA["repos"]["list"] be empty if no data fetched', async () => {
```
[the remark this answers is not in the document]
Comment (Mouhajer-CO) [reply test/src/utils/index.spec.ts:167]: All test start with `should`, lets keep it this way 👍

**Comment by Harrisman05** [comment]
Scripts ran through end-to-end and work. Tests also checked and suggestions made.

**Review by Harrisman05 — APPROVED** [review]

**Pull request merged**